### ISSUE TITLE
Allow users to change the color of their names

### ISSUE THREAD

**Issue by compu42** [vector-im/riot-web#11816]
### Description

I think the user should be able to change the color of their own names (which would be enforced serverside, unless another user's client option overrides it), and I also think that users should be able to force a client side color for all names they see. There should also be an option to choose a random color for each user on the client side. Why should this be done?
- being able to change a color will let users express themselves more
- the random color option will help people identify others faster
- being able to change other colors clientside will make it easier to read for people with cheap/bad monitors, with poor vision and/or in bright environments.
- it's also nice for people who simply don't like the color they see.

Thank you for reading, and I hope you consider my suggestion!

Logs being sent: no, it's not needed

### Version information
- Riot-web: 1.5.6
- OS: Debian 10
- (applies to https://riot.im/app/ and Riot-android Ver. 0.9.9 as well)

**Comment by t3chguy** [comment]
> being able to change other colors clientside will make it easier to read for people with cheap/bad monitors, with poor vision and/or in bright environments.

Most colours people will choose will not play well with themes.

This also begins to feature creep, what if someone wants an animated rainbow name?

**Comment by compu42** [comment]
> > being able to change other colors clientside will make it easier to read for people with cheap/bad monitors, with poor vision and/or in bright environments.
> 
> Most colours people will choose will not play well with themes.

You quoted the clientside part of my suggestion. If the colours don't work with their themes, then they can change it themselves.

> This also begins to feature creep, what if someone wants an animated rainbow name?

we politely tell them no.

**Comment by t3chguy** [comment]
Given you want this enforced server side this is in the wrong repo, it should be in matrix-doc and should probably be structured as an MSC, riot is one client of many for the Matrix protocol.

**Comment by compu42** [comment]
> Given you want this enforced server side this is in the wrong repo, it should be in matrix-doc and should probably be structured as an MSC, riot is one client of many for the Matrix protocol.

Good point. Though, can you tell me what an MSC is? I'm not too familiar with this stuff.

**Comment by t3chguy** [comment]
https://matrix.org/docs/spec/proposals

**Comment by compu42** [comment]
thanks, I'll check that out later today.

Sent from ProtonMail mobile

-------- Original Message --------

**Comment by Kommynct** [comment]
You could limit the colors to a selection, and have that selection change based on whether or not the person is using the dark theme. That'd keep it usable, make it a useful nice bit of personalization, and would allow us to accent things based on their choice of color.

**Comment by krisrobinson** [comment]
> You could limit the colors to a selection, and have that selection change based on whether or not the person is using the dark theme. That'd keep it usable, make it a useful nice bit of personalization, and would allow us to accent things based on their choice of color.

Agree. Limit the pallet but give people the options to change. Half our chat right now is orange and the other is purple, if anything, it's just confusing to what it means, are they mods? why are they purple? Oh its random? We can't change it?

**Comment by KyleMiles** [comment]
I'm just gonna throw my thoughts in here; it's hard to distinguish users in a chat with similar length names of the same color at a glance...handles tend to be pretty static for people across platforms, colors serve as a nice way to distinguish.  Would be nice to have.

In the meantime, is there an admin way to do it on the backend? Wouldn't mind changing it manually for a couple of my users atm.

**Comment by t3chguy** [comment]
No, the colour is never stored, it is a function of the hash of the username.

**Comment by ufgkirk** [comment]
+1 on this feature request... I've had several users on my new Matrix server ask me for this feature.

**Comment by immanuelfodor** [comment]
It would be a relief if I could change the color just for myself, or for anyone in my client with or without backup. I wouldn't care what default color they could see then via id hashing, I just don't like being Mr. Pink :(

**Comment by LoneWolfHT** [comment]
I hate purple. I want a greenish name \:P

**Comment by Panquesito7** [comment]
Sounds like a neat idea 👍 I also wanna change my name color; thought this was possible, as others had other colors. 😄

**Comment by vranki** [comment]
I'd vote against this. Users have different themes and the palette used by Element should be compatible with the used theme. Allowing users to set custom colors would allow unusable colors to be chosen and restricting this would make implementation quite complex.

**Comment by Kommynct** [comment]
By this logic, the current system is also flawed because it chooses colors randomly that don't fit with the theme as well.

**Comment by t3chguy** [comment]
The theme specifies the set of allowed colours. You can create a theme with different colour set.

**Comment by Kommynct** [comment]
Allowing you to choose from the presets does not stop you from creating a theme with a different color set. 

Your green doesn't HAVE to be the same as mine, users don't need to be given full hex control over their colors.

**Comment by immanuelfodor** [comment]
That's why I suggested everybody change colors just on the client side for a current session. This way I could get my favorite color just as I want it, and it wouldn't mess with anyone's theme, it's changing my color only for me. This way I could also change any user's color on the client side, they wouldn't notice it. If this setting would be backed up, it could be transportable between sessions, otherwise, I would be hapy to set it up everytime when I log in for myself.

**Comment by Kommynct** [comment]
porque no las dos?

We could have both, it could be set initially by the user, and you could change it permanently for individual users if you wanted. That'd be easy to add onto the feature, you could even add defaults incase you don't like people having them. Or you want it to be based on their role in the group.

Would that not be fair?

**Comment by mitchnull** [comment]
I've submitted a PR for element-android for part of this issue (overriding the colors by the user on their local clients): https://github.com/vector-im/element-android/pull/2614
If this is accepted, I'll try to add the same to element-web and element-desktop, too.

**Comment by masscream** [comment]
Besides all of the reasons of a kind "I don't like the color for my eyes or/and with the particular theme", I've been asked by one of the users of my matrix-server instance, if it is possible to change the color because she types a lot with someone else whose color looks (almost) same by look and therefore it makes the conversations kind of "messy". It's an accident because both users have totally different username. I went through the settings of Element for the variety of platforms and for desktops I can clearly see a lot of possibilities for customizing the font, size, etc, so color shouldn't be missing imho. Thanks for the consideration and your work.

**Comment by t3chguy** [comment]
Themes can specify the username colors, e.g https://github.com/aaronraimist/element-themes

**Comment by masscream** [comment]
I've realized, that I've probably put my answer onto the wrong place. I've meant to address it more to Element as a global client, my friends I've mentioned use the android version the most. Never mind, the request has been sent that way. Sorry.

**Comment by valdeniomarinho** [comment]
Please add this feature ASAP!

**Comment by kittykat** [comment]
We could do with a minor change to [this PR](https://github.com/matrix-org/matrix-react-sdk/pull/5626) for this issue to be closed. If anyone is interested in adding a commit to make the final changes, that would be really helpful!

**Comment by Alex-Sour** [comment]
This should really be added, otherwise you're stuck with a color, again and again, forever.

**Comment by turt2live** [comment]
As an update: we are very much looking to do this, particularly considering you can on mobile but not web, however the team is stretched a bit too thin to take it on ourselves at the moment. We would really appreciate a PR, likely building off the [existing work](https://github.com/matrix-org/matrix-react-sdk/pull/5626), which adds the functionality so we can get cross-platform compatibility here.

If folks would like to take this on, please do - there's no need to ask for permission ;)

**Comment by Yersi88** [comment]
I would be very excited to see this feature implemented, I'm particularly looking forward to more advanced color schemes perhaps gradient of two colours or different effects. I don't think there's a reason to limit ourselves to one colour when it comes to names.

**Comment by turt2live** [comment]
There's a point where the usability of the app becomes severely affected - I'd encourage PRs which focus on solid colours for now.
